Modules/_ctypes/callproc.c

```diff
@@ -835,15 +836,33 @@ static int _call_function_pointer(int flags,
     if ((flags & FUNCFLAG_CDECL) == 0)
         cc = FFI_STDCALL;
 #endif
-    if (FFI_OK != ffi_prep_cif(&cif,
-                               cc,
-                               argcount,
-                               restype,
-                               atypes)) {
-        PyErr_SetString(PyExc_RuntimeError,
-                        "ffi_prep_cif failed");
-        return -1;
+
+#if defined(__APPLE__) && __arm64__
+    if (argtypecount != 0 && argcount > argtypecount) {
```

Comment:
Is this test correct?

<http://www.chiark.greenend.org.uk/doc/libffi-dev/html/The-Basics.html> indicates that ffi_prep_cif_var is not the same as ffi_prep_cif even if argtypecount == argcount.

Reply:
"Is this test correct?"

It's not correct, but it is less incorrect than it was before.

"The iOS ABI for functions that take a variable number of arguments is entirely different from the generic version."

That's the same ABI as arm64 Mac OS.

https://developer.apple.com/library/archive/documentation/Xcode/Conceptual/iPhoneOSABIReference/Articles/ARM64FunctionCallingConventions.html

But the problem is, I don't see a way to specify a function to be variadic in the ctypes API, and lots of existing code out there isn't going to know to specify it even if we add it.   They're just going to set the arg types and call the function with extra args and expect it to work.

This change at least cover the typical case where there is at least one variable arg passed to the function.     

Should we add a new documented feature to ctypes allowing the user to specify a function is variadic?